### ISSUE TITLE
Roberta model will be different after model.to(xla_device) in TPU training

### ISSUE THREAD

**Issue by fe1ixxu** [pytorch/fairseq#3449]
## ❓ Questions and Help
I was following the Roberta training tutorial (https://cloud.google.com/tpu/docs/tutorials/roberta-pytorch) to train a Roberta model from scratch. After training 100K steps, I tried to load checkpoint (`RobertaModel.from_pretrained()`) to do fill mask tasks. The results look randomly generated but the loss of training is ~ 2.7, so I took a closer look at the code. I found that if I initialized a Robertamodel -> send it to xla (model.to(xla_device))  -> load the checkpoint to this model -> fill mask task, everything looks fine. I post the results below:

Sentence: As a [MASK], she feels that publishing papers is more difficult than what she thought:
1. Load model with the original way:  initialize a Robertamodel -> load checkpoint: 
   The top 5 candidates will be ['[UNK]', 'felony', 'proceeding', 'bodily', '##█']
2. The second way: initialize a Robertamodel -> model.to(xla_device) -> load checkpoint:
   The  top 5 candidates will be ['writer', 'result', 'student', 'woman', 'journalist']
3. The third way: initialize a Robertamodel -> model.to(xla_device) - > model.to("cpu") -> load checkpoint:
   The  top 5 candidates will be ['writer', 'result', 'student', 'woman', 'journalist']

so the model will be somehow different after sending it to xla?? I have checked the architectures of both models and they are the same.

I also did a small experiment: I saved (torch.save) an initialized Roberta model in cpu and also saved the model after feeding it to xla. Then they have different sizes...

What is going on...? How can I just correctly load the checkpoint without sending it to xla device??
